src/software/thunderscope/gl/helpers/gl_patches.py

```diff
@@ -0,0 +1,71 @@
+from pyqtgraph.opengl.GLGraphicsItem import GLGraphicsItem
+from pyqtgraph.opengl.GLViewWidget import GLViewMixin
+
+
+def GLGraphicsItem_setParentItem_patched(self, parent: GLGraphicsItem) -> None:
+    """Patched version of GLGraphicsItem.setParentItem that properly
+    removes this item from the scenegraph when the parent param is None.
+
+    From the original pyqtgraph documentation:
+    Sets this item's parent in the scenegraph hierarchy.
+
+    :param parent: the GLGraphicsItem to set as the parent of this item.
+                   If None, this item is removed from the scenegraph.
+    """
+    if self._GLGraphicsItem__parent:
+        self._GLGraphicsItem__parent._GLGraphicsItem__children.remove(self)
+    self._GLGraphicsItem__parent = None
+    if self.view():
+        self.view().removeItem(self)
+
+    if parent:
+        parent._GLGraphicsItem__children.add(self)
+        self._GLGraphicsItem__parent = parent
+        if parent.view():
+            parent.view().addItem(self)
+
+
+def GLViewMixin_addItem_patched(self, item: GLGraphicsItem) -> None:
+    """Patched version of GLViewMixin.addItem that properly adds all
+    the item and all its descendants to the scene.
+
+    :param item: the item to add to the scene
+    """
+    items_to_add = [item]
+
+    while items_to_add:
+        item_to_add = items_to_add.pop()
+        self.items.append(item_to_add)
+        item_to_add._setView(self)
+
+        if self.isValid():
+            item_to_add.initialize()
+
+        for child in item_to_add.childItems():
+            items_to_add.append(child)
+
+    self.update()
+
+
+def GLViewMixin_removeItem_patched(self, item: GLGraphicsItem) -> None:
+    """Patched version of GLViewMixin.removeItem that properly removes
+    the given item and its descendants from the scene.
+
+    :param item: the item to remove from the scene
+    """
+    items_to_remove = [item]
+
+    while items_to_remove:
+        item_to_remove = items_to_remove.pop()
+        self.items.remove(item_to_remove)
+        item_to_remove._setView(None)
+
+        for child in item_to_remove.childItems():
+            items_to_remove.append(child)
+
+    self.update()
+
+
+GLGraphicsItem.setParentItem = GLGraphicsItem_setParentItem_patched
+GLViewMixin.addItem = GLViewMixin_addItem_patched
+GLViewMixin.removeItem = GLViewMixin_removeItem_patched
```

Comment:
Did not know you could do something like that in python. 

Also, does the order of importing now matter? 

So will 

```
import <long_list_of_stuff>/gl_patches 
import pyqtgraph
```

verus
```
import pyqtgraph
import <long_list_of_stuff>/gl_patches 
```
matter?

Reply:
Problem not something you need to worry about. As long as the `import ... gl_patches` happens before the `GLViewWidget` or any `GLGraphicsItem`s are created, the patch should work